documentation/docs/contribute/development-guide/coding/01-architecture.md

```diff
@@ -95,25 +95,32 @@ We follow the [Clean Architecture](https://blog.cleancoder.com/uncle-bob/2012/08
 
 It is a layered approach for a [more civilized age](https://www.youtube.com/watch?v=wtCQalq7L-E).
 
+Clean architecture aims to divide a big project into layers and pieces (classes, entities, modules etc.), and to make them independent of each other, so that our big and complex project become more maintainable, flexible and easy to work on together.
+
 ![Clean Architecture](/img/clean-architecture2.png)
 
-The main point of this approach is that the business logic, also known as domain, is at the center of the universe.
+The key idea behind this diagram is that the application should be divided into layers (there can be any amount of layers). The inner layers do not know about outer ones, the dependencies are directed to the center. The far a layer is from the center, the more it knows about "non-business" details of the application (e.g. which framework is used).
 
-#### The Dependency Rule
+The idea to keep the core business logic (entities) and application logic (use cases) at the center of the solution structure, and independent of frameworks, databases, and external parts (the two outer layers). This is to allow the core part, adaptive and flexible enough to deal with changing technology and interfaces, so that frameworks, databases, and external parts can be replaced with minimum efforts.
```

Comment:
```suggestion
The idea to keep the core business logic (entities) and application logic (use cases) at the center of the solution structure, and independent of frameworks, databases, and external parts (the two outer layers). This allows the core part to be adaptive and flexible enough to deal with changing technology and interfaces, so that frameworks, databases, and external parts can be replaced with minimum efforts.
```

Reply:
I wasn't sure how to write it, but I think I would point out that not only can we replace FastAPI with Flask, but we can actually suddenly develop a CLI for our application without rewriting any of our core business logic.

Reply:
that is correct :)

Reply:
I see you mentioned it in the dependency rule section, so it's not SO critical.

Reply:
True, it is a bit more in depth there. Fixed comment.